### ISSUE TITLE
shader compilation error, crash after creating game project and can't import .project files

### ISSUE THREAD

**Issue by fgsoftware1** [antopilo/Nuake#60]
shader compilation error: 
```cmd
[17:02:02] renderer - Driver detected 4.3.0 - Build 20.19.15.4531
[17:02:02] texture - Failed to load texture: resources/Textures/default/Default.png
[17:02:02] texture - Failed to load texture: resources/Textures/default/defaultNormal.png
[17:02:02] texture - Failed to load texture: default
Failed to compile vertex shader!
ERROR: 0:2: '' :  incorrect GLSL version: 460
WARNING: 0:2: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:2: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:3: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:3: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:3: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:4: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
ERROR: 0:8: 'vec2' : syntax error syntax error


Failed to compile Fragment shader!
ERROR: 0:2: '' :  incorrect GLSL version: 460
WARNING: 0:22: 'assign' : implicit type conversion allowed from GLSL 1.20
WARNING: 0:22: 'assign' : implicit type conversion allowed from GLSL 1.20
WARNING: 0:27: 'assign' : implicit type conversion allowed from GLSL 1.20
WARNING: 0:27: 'assign' : implicit type conversion allowed from GLSL 1.20
WARNING: 0:32: 'assign' : implicit type conversion allowed from GLSL 1.20
WARNING: 0:32: 'assign' : implicit type conversion allowed from GLSL 1.20
WARNING: 0:57: 'texture' : function is not available in current GLSL version


Failed to compile vertex shader!
ERROR: 0:2: '' :  incorrect GLSL version: 460
WARNING: 0:2: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:2: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:3: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:3: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:3: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:4: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
ERROR: 0:8: 'vec2' : syntax error syntax error


Failed to compile Fragment shader!
ERROR: 0:2: '' :  incorrect GLSL version: 460
ERROR: 0:4: 'vec2' : syntax error syntax error


[17:02:02] main - Engine initialized
Failed to compile vertex shader!
ERROR: 0:2: '' :  incorrect GLSL version: 460
WARNING: 0:3: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:3: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:4: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:4: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:4: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:5: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version


Failed to compile Fragment shader!
ERROR: 0:2: '' :  incorrect GLSL version: 460
```

editor crash after creating a project.
can't import .project files(probably mistake: it appears allowed files to be imported are |*.project, .project files don't appear on import screen)

**Comment by fgsoftware1** [comment]
that problems on release

**Comment by antopilo** [comment]
Hello, thanks for checking the project out!

Is this on Windows?

**Comment by fgsoftware1** [comment]
> Hello, thanks for checking the project out!
> 
> Is this on Windows?

yes

**Comment by fgsoftware1** [comment]
github realeases version @antopilo

**Comment by fgsoftware1** [comment]
@antopilo i'll compile and run to see if happens the same and return to you

**Comment by antopilo** [comment]
Indeed, there is some kind of error with the *release* version at the moment. 

I will need to investigate it.

Are you able to compile it yourself?

**Comment by fgsoftware1** [comment]
> Indeed, there is some kind of error with the _release_ version at the moment.
> 
> I will need to investigate it.
> 
> Are you able to compile it yourself?

i'm updating the proj, it will take a while to test but i'll return to you when i get it done

**Comment by fgsoftware1** [comment]
@antopilo but as my knowledge, you're using things on shading code that aren't supported by that version, the crashing i don't really know, that project import problem i think i can help with

**Comment by antopilo** [comment]
What is your GPU?

**Comment by fgsoftware1** [comment]
> What is your GPU?

intel hd family, but this shouldn't be a problem cause it found the driver, idk

**Comment by antopilo** [comment]
It seems like your integrated gpu doesn't support the required opengl version, I'll see if I can downgrade the current opengl version

**Comment by fgsoftware1** [comment]
> It seems like your integrated gpu doesn't support the required opengl version, I'll see if I can downgrade the current opengl version

let me compile and see how i goes

**Comment by fgsoftware1** [comment]
@antopilo compiled sucessfully, opengl errors still, crash when project created still, the problem of .project not appearing fixed and import as well but the editor is like hidden(it appears on task bar, i can see if i hover it but i can't pull it to the screen)

**Comment by fgsoftware1** [comment]
@antopilo couldn't figure out my GPU model very well but it seems to support opengl 4.3

**Comment by fgsoftware1** [comment]
@antopilo i wonder why you're using this opengl version since imgui runs on opengl3 and glfw default is 3 as well

**Comment by antopilo** [comment]
Fixed with commit b2656650133cbfef14d2007e96d0623bdc17b81c
